waggle-dance-api/src/main/java/com/hotels/bdp/waggledance/api/model/PrimaryMetaStore.java

```diff
@@ -16,13 +16,16 @@
 package com.hotels.bdp.waggledance.api.model;
 
 import java.util.Arrays;
+import java.util.Collections;
 import java.util.List;
 
 import javax.validation.constraints.NotNull;
 
 public class PrimaryMetaStore extends AbstractMetaStore {
 
   private static final String EMPTY_PREFIX = "";
+  private @NotNull List<String> mappedDatabases = Collections.emptyList();
```

Comment:
I still think it can be null as it is an optional thing. The default should be null which is accept all. It shouldn't be emptyList which means map nothing.